buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/VerifyBuildEnvironmentPlugin.kt

```diff
@@ -0,0 +1,49 @@
+package org.gradle.gradlebuild.buildquality
+
+import availableJavaInstallations
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import java.nio.charset.Charset
+import org.gradle.api.tasks.compile.AbstractCompile
+import org.gradle.kotlin.dsl.*
+
+
+open class VerifyBuildEnvironmentPlugin : Plugin<Project> {
+    override fun apply(project: Project): Unit = project.run {
+        validateForAllBuilds(project)
+        validateForProductionEnvironments(project)
+        validateForAllCompileTasks(project)
+    }
+
+    private
+    fun validateForAllBuilds(rootProject: Project) = rootProject.afterEvaluate {
+        rootProject.availableJavaInstallations.validateForAllBuilds()
```

Comment:
This should only verify that the versions are set up correctly, right? We can do it then during configuration time.